feat: move eventyay-video services, serializers, and views to eventyay base (attendee UI excluded)

suhailnadaf509 opened the pull request:
Resolves #753

## Summary by Sourcery

Migrate the video subsystem from the `venueless` package into the new `eventyay.base` namespace while retaining full functionality.

Enhancements:
- Move all management commands, services, serializers, views, and utilities from `video/server/venueless` to `eventyay/base` and update import paths accordingly
- Extend the base models module by importing additional entities (polls, posters, questions, rooms, roulette, streaming, turn server, world) and updating the public `__all__` export
- Rename Django app configurations and labels from `venueless` to `eventyay.base` and adjust template branding
- Update JWT audience settings and StatsD namespace from `venueless` to `eventyay`

Sak1012 reviewed (changes requested):
Please align the file structure with the existing `Tickets` convention. Since we already have `/api` and `/control` directories at the root level, there's no need to include them inside the `/base` folder. Additionally, ensure that all related dependency files are moved to their appropriate directories to maintain a clean and organized structure.

suhailnadaf509 commented:
@Sak1012 I've moved the files from eventyay/base/control to eventyay/control but couldn't do the same for /api as it lead to many circular import errors regarding auth ,if needed i could rename the folder to avoid confusion

suhailnadaf509 commented:
@Sak1012 I've added some of the requested changes such as 
1. unified API directory (got rid of the circular imports issue )
2. merged the services from core with the services in the base
3. moved live , analytics , social , importers and integrations out of base and into a new folder called features under /eventyay
will now be working on more changes such as making a unified templates folder , combing the utils , and trying to desolve the core directory , maybe also look into auth etc 
let me know if there are any changes needed

suhailnadaf509 commented:
@Sak1012  I've modified the user model of tickets by extending the fields to have the world, tokens logic of video . I have modified the imports to depend on the unified auth file(/base/models) 
Resolves #778

Sak1012 commented on eventyay/control/models.py:
Move model to base

suhailnadaf509 replied:
Moved the model to base and renamed it to control_model.py

Sak1012 commented on eventyay/control/forms.py:
Control will have multiple forms please rename this and add under `/forms`

suhailnadaf509 replied:
I have placed this under forms and renamed this to server_management.py

Sak1012 commented on eventyay/control/static/control/jquery-3.6.0.js:
Move the static files to a separate directory eventyay/static

suhailnadaf509 replied:
This has been done!

Sak1012 commented on eventyay/control/views.py:
Controls will also have a lot more views move this file into /control after renaming with the respective views.

Sak1012 commented on eventyay/core/utils/config.py:
```diff
@@ -2,14 +2,14 @@
-    Room,
-    World,
 )
+from eventyay.base.models.room import Room
```
These can be imported the same way as the other models.

suhailnadaf509 replied:
They can't be imported the same way as the before ( pointing to init.py ) because that leads to a circular import issue so i had to directly point to the model file itself

Sak1012 commented on eventyay/core/models/__init__.py:
Add the file from video component and make the changes.

suhailnadaf509 replied:
This has been done!

suhailnadaf509 commented on eventyay/api/serializers/serializers.py:
```diff
@@ -1,8 +1,8 @@
+from eventyay.base.models.room import Room
 
 
 class RoomSerializer(serializers.ModelSerializer):
```
I have renamed the file from serializers.py to rooms.py and changed the imports accordingly

suhailnadaf509 commented on eventyay/base/storage/models.py:
I have pushed storage out of base and into the root directory(/eventyay) and placed models.py from storage to base/models and renamed the file to storage_model.py

suhailnadaf509 commented:
@Sak1012 I've migrated the codebase from "world" to "event" :

- I've added the missing fields and models from world(base/models) to event(base/models)
- Updated all models, services, APIs, and views , changed URLs from /worlds/ to /events/ 

places like services and custom made functions related to world where i was unable to find replacement with event so there i have just renamed the references

Sak1012 commented on eventyay/base/models/control_model.py:
Is this file required, The whole file is commented out.

suhailnadaf509 replied:
The file is no longer required, as the model "LogEntry" is already in tickets , will be sending this file 
back to /video

hongquan reviewed (changes requested):
Please get your branch updated with the `enext` branch.

hongquan commented:
@suhailnadaf509 The changes are so big that we need to see some screenshots proving that it is working.

suhailnadaf509 commented:
> With only screenshots for the "control/admin" area, your PR title should be smaller scope, it is not "full functionality" at all.

@hongquan  This PR merges the majority of Venueless functionality into /eventyay. The /control/admin view remains the only one served directly by Venueless, marking the only UI distinction; all remaining views are now handled within the webapp.
